results/cmd/api/main_test.go

```diff
@@ -61,5 +65,46 @@ func setupTestDB(dbName string, t *testing.T) (*sql.DB, error) {
 	if _, err := statement.Exec(); err != nil {
 		t.Fatalf("failed to execute the taskrun table creation statement statement: %v", err)
 	}
-	return db, nil
+	return srv, nil
+}
+func TestUpdateTaskRun(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB("testdb", t)
+	if err != nil {
+		t.Fatalf("failed to create temp file for db: %v", err)
+	}
+
+	// Connect to fake server and create a taskrun
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	objectmeta := &pb.ObjectMeta{Uid: "123459", Name: "mytaskrun", Namespace: "default"}
+	r, err := srv.CreateTaskRun(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "1", Metadata: objectmeta}})
+	if err != nil {
+		t.Fatalf("could not create taskrun: %v", err)
+	}
+
+	// Update the created taskrun
+	objectmeta = &pb.ObjectMeta{Uid: "123459", Name: "ziweifan", Namespace: "tomorrow"}
+	r, err = srv.UpdateTaskRun(ctx, &pb.UpdateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1alpha1", Metadata: objectmeta}})
+	if err != nil {
+		t.Fatalf("could not update taskrun: %v", err)
+	}
+
+	// Validate by checking if we can get the updated taskrun
+	rows, err := srv.db.Query("SELECT taskrunlog FROM taskrun WHERE uid = ?", r.GetMetadata().GetUid())
+	if err != nil {
+		t.Fatalf("failed to query on database: %v", err)
+	}
+	for rows.Next() {
+		var taskrunblob []byte
+		taskrun := &pb.TaskRun{}
+		rows.Scan(&taskrunblob)
+		if err = proto.Unmarshal(taskrunblob, taskrun); err != nil {
+			t.Fatal("unmarshaling error: ", err)
+		}
+		metadata := taskrun.GetMetadata()
+		if metadata.GetName() != r.GetMetadata().GetName() || metadata.GetNamespace() != r.GetMetadata().GetNamespace() {
+			t.Fatal("Update Function not properly implemented.")
+		}
```

Comment:
Use https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc#Diff instead to compare the entire struct.

Reply:
Done